### ISSUE TITLE
Investigate if this code change makes sense

### ISSUE THREAD

**Issue by hgducharme** [hgducharme/meatball#37]
We have the following:

```
Bitboard calculatePawnAttacksFromSquare(const Color color, const Bitboard & bitboard)
{
    Bitboard potentialPawnAttacks;
    Rank rank = Chessboard::squareToRank(bitboard.findIndexLSB());

    // A pawn only has valid moves between ranks 2 and 7
    if (rank == RANK_1 || rank == RANK_8)
    {
        return potentialPawnAttacks;
    }

    // White pawns can only move north and black pawns can only move south
    Direction eastDirection = NORTH_EAST;
    Direction westDirection = NORTH_WEST;
    if (color == BLACK)
    {
        eastDirection = SOUTH_EAST;
        westDirection = SOUTH_WEST;
    }

    // If we perform a move and end up on the opposite side of the board, that is an off-board move and we need to exclude that move
    potentialPawnAttacks |= utils::shiftCurrentSquareByDirection(bitboard, eastDirection) & constants::bit_masks::EXCLUDE_FILE_A;
    potentialPawnAttacks |= utils::shiftCurrentSquareByDirection(bitboard, westDirection) & constants::bit_masks::EXCLUDE_FILE_H;

    return potentialPawnAttacks;
}
```

But the inside code doesn't even need `bitboard`. It just needs the square as an integer value. Maybe we can do this:

```
Bitboard calculatePawnAttacksFromSquare(const Color color, const Square square)
{
    Bitboard potentialPawnAttacks;
    Rank rank = Chessboard::squareToRank(square);

    ...
}
```

There are many such changes like this to be made in the file `attacktables.cpp`

**Comment by hgducharme** [comment]
I do the same thing here:

```
Bitboard shiftCurrentSquareByDirection(const Square square, const int numberOfBits)
{
   Bitboard squareAsBitboard(square);
   return shiftCurrentSquareByDirection(squareAsBitboard, numberOfBits);
}

Bitboard shiftCurrentSquareByDirection(const Bitboard & oldPosition, const int numberOfBits)
{
   Bitboard newPosition;

   if (numberOfBits > 0)
   {
      newPosition = oldPosition << numberOfBits;
   }
   else
   {
      newPosition = oldPosition >> std::abs(numberOfBits);
   }

   return newPosition;
}
```

Note that the first example in the [CPP Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#p1-express-ideas-directly-in-code) explicitly says not to do this

**Comment by hgducharme** [comment]
Ok so I implemented the following change:

```cpp
// utils.h
Bitboard shiftSquareByDirection(const Square square, const int numberOfBits);
Bitboard shiftSquareByDirection(const Bitboard & oldPosition, const int numberOfBits);
```

became:

```cpp
// utils.h
Bitboard getSquareInDirectionAsBitboard(const Square square, const int numberOfBits);
```

That is, I changed the name of the function to be more precise and deleted the overload that took a Bitboard argument. Now all calls to this function pass in a Square, and it alleviates the concerns of this issue.

**Comment by hgducharme** [comment]
Closed by [b5d3b0e](https://github.com/hgducharme/meatball/pull/56/commits/b5d3b0ebe40561c4313a2e036a6f7d1534c16e35) in #56
